backend/api/Areas/Tools/Helpers/ImportPropertiesHelper.cs

```diff
@@ -247,19 +248,45 @@ public IEnumerable<Entity.Parcel> AddUpdateProperties(IEnumerable<Model.ImportPr
 
                 if (String.Compare(property.PropertyType, "Land") == 0)
                 {
+                    // first check to see if there is an existing parcel with the pid in the database
+                    var isPidAvailable = _pimsAdminService.Parcel.IsPidAvailable(pid);
+
                     entities.Add(AddUpdateParcel(property, pid, agency));
+
+                    // then set whether the property was updated or added based on whether the parcel was already in database or not                    
+                    if (isPidAvailable)
+                    {
+                        property.Added = true;
+                        property.Updated = false;
+                    }
+                    else
+                    {
+                        property.Added = false;
+                        property.Updated = true;
+                    }
+                    propertiesAddedOrEdited.Add(property);
                 }
                 else if (String.Compare(property.PropertyType, "Building") == 0)
                 {
-                    var parcel = AddUpdateBuilding(property, pid, agency);
-                    if (!entities.Any(p => p.PID == parcel.PID))
+                    var isBuildingExisting = _pimsAdminService.Building.GetByNameAddressWithoutTracking(property.Name, property.CivicAddress);
+                    // need to check the count before a building gets added, as the previous variable gets updated once the building has been added
```

Comment:
Created a new function to check for an existing building before adding or updating the building.